gcsio/src/main/java/com/google/cloud/hadoop/gcsio/GoogleCloudStorageClientReadChannel.java

```diff
@@ -494,6 +501,9 @@ private BlobSourceOption[] generateReadOptions(BlobId blobId) {
       if (blobId.getGeneration() != null) {
         readOptions.add(BlobSourceOption.generationMatch(blobId.getGeneration()));
       }
+      if (requesterPays) {
+        readOptions.add(BlobSourceOption.userProject(storageOptions.getRequesterPaysOptions().getProjectId()));
```

Comment:
is it ok to mutate this? If this object is getting reused, the next Object will also have value set, correct?

Mutating the value of the input parameter is an anti-pattern.